### PR TITLE
medic check

### LINKED ISSUE #6: [SUGGESTION] Tell bleeding out player whether EMS is on duty
**Is your feature request related to a problem? Please describe.**
A player bleeding out or has already bled out may be ok waiting for EMS to arrive, but does not know whether anyone is coming or not.

**Describe the feature you'd like**
The 3d text that displays to the downed user should call qb-core to check whether any EMS are on duty and relay that info to the player.

**Describe alternatives you've considered**
Not showing the option to call EMS if none are online, but this has some edge cases. Like what happens if an EMS comes online during, or goes offline after they are called.

### PR DESCRIPTION
**Pull request by YvanCywan**
**Describe Pull request**
To Close #6
Including a medic check to ensure that the player know if there are any medics online or not.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [no] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

**Comment by Manason** [comment]
This doesn't actually work for the edge case of an EMS goes offline after you've called them?

File: client/dead.lua [hunk 1]
```diff
@@ -4,6 +4,14 @@ local hold = 5
 deathTime = 0
 
 -- Functions
+local function MedicCheck()
```
Comment (Manason) [inline client/dead.lua]: Consider renaming to "IsEmsOnDuty"

File: client/dead.lua [hunk 2]
```diff
@@ -4,6 +4,14 @@ local hold = 5
...
+    QBCore.Functions.TriggerCallback('hospital:GetDoctors', function(medics)
+        if medics < 1 then
+            return false
+        end
+        return true
```
Comment (Manason) [inline client/dead.lua]:
```suggestion
        return medics > 0
```

File: client/dead.lua [hunk 3]
```diff
@@ -186,8 +194,12 @@ CreateThread(function()
...
 
                     if IsControlJustPressed(0, 47) and not emsNotified then
-                        TriggerServerEvent('hospital:server:ambulanceAlert', Lang:t('info.civ_down'))
-                        emsNotified = true
+                        if not MedicCheck() then
+                            lib.notify({ type = 'error', message = Lang:t('error.no_medics') })
+                        else
+                            TriggerServerEvent('hospital:server:ambulanceAlert', Lang:t('info.civ_down'))
+                            emsNotified = true
+                        end
```
Comment (Manason) [inline client/dead.lua]: Consider changing to if MedicCheck() to get rid of the not and doing the second condition first to reduce cognitive complexity.

**Review by Manason — APPROVED** [review]

**Comment by HebiKotei** [comment]
> This doesn't actually work for the edge case of an EMS goes offline after you've called them?

It should since the thread is one big loop. When the loop refreshes, it should register the previously on duty EMS to off duty, which then should throw the error. The button to call EMS does refresh allowing another click.

**Comment by YvanCywan** [comment]
> This doesn't actually work for the edge case of an EMS goes offline after you've called them?

I've sorted this edge case now, placed a check before the button press, the error notification should disappear/reappear as when EMS is on the server. This also works on each script update.

File: client/dead.lua [hunk 4]
```diff
@@ -185,9 +190,15 @@ CreateThread(function()
                         DrawTxt(0.90, 1.40, 1.0, 1.0, 0.6, Lang:t('info.help_requested'), 255, 255, 255, 255)
                     end
 
+                    if not IsEmsOnDuty() then
```
Comment (ChatDisabled) [inline client/dead.lua]: Doesn't this get spammed since this is inside of a loop? Don't think players want a screen filled with notify

Comment (YvanCywan) [reply client/dead.lua]: This is fair, my workaround in the next commit will be to implement a check directly into the respawn timer, enabling you to just press the respawn button provided that there is no EMS on duty.

File: client/dead.lua [hunk 5]
```diff
@@ -186,8 +193,10 @@ CreateThread(function()
                     end
 
                     if IsControlJustPressed(0, 47) and not emsNotified then
-                        TriggerServerEvent('hospital:server:ambulanceAlert', Lang:t('info.civ_down'))
-                        emsNotified = true
+                        if IsEmsOnDuty() then
```
Comment (Manason) [inline client/dead.lua]: consider combining these two ifs using an and.

File: client/dead.lua [hunk 6]
```diff
@@ -145,8 +150,10 @@ CreateThread(function()
 
             if isDead then
                 if not isInHospitalBed then
-                    if deathTime > 0 then
+                    if deathTime > 0 and IsEmsOnDuty() then
                         DrawTxt(0.93, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_txt', { deathtime = math.ceil(deathTime) }), 255, 255, 255, 255)
+                    elseif not IsEmsOnDuty() then
+                        DrawTxt(0.865, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_revive', { holdtime = hold, cost = Config.BillCost }), 255, 255, 255, 255)
                     else
                         DrawTxt(0.865, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_revive', { holdtime = hold, cost = Config.BillCost }), 255, 255, 255, 255)
```
Comment (D4isDAVID) [inline client/dead.lua]:
```suggestion
                    else
                        DrawTxt(0.865, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_revive', { holdtime = hold, cost = Config.BillCost }), 255, 255, 255, 255)
```
What's the point of this? This is just duplicate code.

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

File: client/dead.lua (hunk 6 above)
Comment (YvanCywan) [reply client/dead.lua]: Urite

**Review by D4isDAVID — APPROVED** [review]

**Pull request merged**